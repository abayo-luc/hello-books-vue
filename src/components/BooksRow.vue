<template>
  <div class="row">
    <book-card v-for="book in books" :key="book.id" :book="book" />
  </div>
</template>

<script>
import BookCard from './cards/BookCard.vue';

export default {
  name: 'BooksRow',
  props: {
    books: {
      type: Array,
      required: true,
    },
  },
  components: {
    BookCard,
  },
};
</script>

<style scoped>
</style>
