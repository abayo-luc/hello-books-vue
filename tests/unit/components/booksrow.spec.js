import {
  mount,
} from '@vue/test-utils';
import BooksRow from '../../../src/components/BooksRow.vue';

const wrapper = mount(BooksRow, {
  propsData: {
    books: [{
      rating: 8,
      author: 'Chinua Achebe',
      title: 'Things Fall Apart',
      genre: 'Novel',
      url: 'https://via.placeholder.com/600/92c952',
    }],

  },
});

describe('BookCard.vue', () => {
  it('should render book title', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
