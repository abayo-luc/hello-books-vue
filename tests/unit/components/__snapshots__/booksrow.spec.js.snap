// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookCard.vue should render book title 1`] = `
<div class="row">
  <div class="book-card"><img src="https://via.placeholder.com/600/92c952" class="book-cover">
    <div class="content">
      <h2 class="rating">6.8</h2>
      <h3 class="book-title">Things Fall Apart</h3>
      <p class="author">Joanne Ramos</p>
      <div class="category">
        <p>Novel</p>
      </div>
    </div>
  </div>
</div>
`;
